src/WebJobs.Script.Grpc/Channel/GrpcCapabilities.cs

```diff
@@ -26,18 +26,34 @@ public string GetCapabilityState(string capability)
             return null;
         }
 
-        public void UpdateCapabilities(IDictionary<string, string> capabilities)
+        public void UpdateCapabilities(IDictionary<string, string> capabilities, GrpcCapabilitiesUpdateStrategy strategy)
         {
             if (capabilities == null)
             {
                 return;
             }
 
-            _logger.LogDebug($"Updating capabilities: {capabilities.ToString()}");
+            if (strategy == GrpcCapabilitiesUpdateStrategy.Merge)
+            {
+                _logger.LogDebug($"Updating capabilities using merge strategy: {capabilities.ToString()}");
```

Comment:
Couple of nits about the logging:

1. You can do this with only one message by using `strategy.ToString()` and moving the log up before we enter the if/else (makes it nicer if we add new strategies)
2. Can we be more explicit here? With something like `"Updating capabilities using {strategy} strategy. Current: {_capabilities} Incoming: {_capabilities}"`
3. You shouldn't need `ToString()` inside an interpolated string... it will do that automatically (I know you didn't add it, but let's clean it up a little :-))
4. And then we can have another message after another message `"Updated capabilities: {_capabilities.ToString()}"`

That would help debug some of this should there be issues.